crates/sui-node/src/lib.rs

```diff
@@ -388,11 +417,40 @@ impl SuiNode {
             .ok_or_else(|| anyhow::anyhow!("Transaction Orchestrator is not enabled in this node."))
     }
 
-    //TODO watch/wait on all the components
-    pub async fn wait(self) -> Result<()> {
-        self.grpc_server.await??;
-
-        Ok(())
+    pub async fn monitor_reconfiguration(mut self) -> Result<()> {
+        loop {
+            let next_epoch = self
+                .reconfig_channel
+                .1
+                .recv()
+                .await
+                .expect("Reconfiguration channel was closed unexpectedly.");
+            info!(
+                ?next_epoch,
+                "Received reconfiguration signal. About to reconfigure the system."
+            );
+            let new_committee = self
+                .state
+                .get_sui_system_state_object()
+                .await
+                .unwrap()
+                .get_current_epoch_committee();
+            if let Some((_, _)) = &self.validator_server_info {
+                // TODO: Shutdown Narwhal here.
+                self.state.reconfigure(new_committee.committee).unwrap();
+                // TODO: Start Narwhal here.
+            } else if self.state.is_validator() {
+                self.validator_server_info = Self::start_grpc_validator_service(
+                    &self.config,
+                    self.state.clone(),
+                    self.checkpoint_store.clone(),
+                    &self._state_sync,
+                    &self._prometheus_registry,
+                )
+                .await
+                .unwrap();
+            }
```

Comment:
When do we expect this to happen? I mean, wouldn't the grpc validator service running already?

Reply:
When a node starts up, it may not be a validator yet (i.e. it's not part of the current committee), so it's essentially a fullnode. In that case we didn't start the grpc service. 
Then after a reconfiguration, it becomes a validator.
This could be an overkill, though. We could choose to always run the grpc server. It probably doesn't hurt.

Reply:
Ah I see. As far as I can tell the `start_grpc_validator_service` is spins up the `ValidatorService` which in turns runs a bunch of stuff (Narwhal amongst them). So I would say your approach here seems very valid - otherwise we might have a node that is trying to do stuff it supposed not to (ex trying to connect to other NW nodes etc)

Reply:
We won't be starting the server again if we have already started it before.